python/nav/etc/alertmsg/deviceState/deviceDeletedFan-email.txt

```diff
@@ -0,0 +1,2 @@
+Subject: Device has been removed as a fan
+Device {{ device.get_extended_description }} has been removed from {{netbox}} as a fan
```

Comment:
Been a long time since i made this so honestly not sure, but that sounds likely. It should still be fine, since `get_extended_description` checks if a related object exists or not. If there is no related object it uses the serial number (or if no serial number, empty string), so its fine to use either way

Reply:
I suppose it would be slightly faster to assume that there is no "extended description" in this case, and just use the device str representation directly, but i dont have strong feelings either way

Reply:
I use the same `get_extended_description`in #2998, and heres an example of the message generated:

> Device Module EX3400-48P at sikt-osl-3380-sw1.c.uninett.no has been removed from sikt-osl-3380-sw1.c.uninett.no as a module 

So in that instance it seems to have been able to find the `Module` objects somehow, but the situation might be different here

Reply:
I do suspect this could work just due to random timing: If the event gets committed separately, while the actual delection gets committed later, eventengine may have time to react to the event before the module is actually removed (it reacts to events using PostgreSQL LISTEN/NOTIFY mechanism).

It could be interesting to see what happens if you turn off the event engine (simulating some downtime) and restart it only after the ipdevpoll job is completed.  What does it make of the event at that time?


Anyway, your example does show some ridiculous redundancies in the resulting alert message (the netbox name is mentioned twice).  So yeah, it might actually be simpler to just use the `str` representation of the `Device`, which should default to its serial number.  The rest of the message already specifies that it used to be a fan or a PSU in netbox so-and-so.

Reply:
ye, the naming is a bit crazy. I reckon this is the case for alot of alerts. Plenty of templates use both `get_extended_description` and the netbox name, resulting in duplicate netbox naming.

I'll go for device directly in both PRs

Reply:
> Plenty of templates use both `get_extended_description` and the netbox name, resulting in duplicate netbox naming.

If that's the case, I think you should post an issue pointing them out so they can be fixed :)